src/main/kotlin/com/openlattice/assembler/AssemblerConnectionManager.kt

```diff
@@ -470,21 +502,34 @@ class AssemblerConnectionManager(
             principal: Principal,
             columns: List<String>
     ): String {
-        val postgresUserName = if (principal.type == PrincipalType.USER) {
-            buildPostgresUsername(securePrincipalsManager.getPrincipal(principal.id))
-        } else {
-            buildPostgresRoleName(securePrincipalsManager.lookupRole(principal))
-        }
+        val postgresUserName =
+                if (principal.type == PrincipalType.USER) {
+                    buildPostgresUsername(securePrincipalsManager.getPrincipal(principal.id))
+                } else {
+                    buildPostgresRoleName(securePrincipalsManager.lookupRole(principal))
+                }
+
+        return grantSelectSql(entitySetTableName, quote(postgresUserName), columns)
+    }
 
+    /**
+     * Build grant select sql statement for a given table and user with column level security.
+     * If properties (columns) are left empty, it will grant select on whole table.
+     */
+    private fun grantSelectSql(
+            tableName: String,
+            postgresUserName: String,
+            columns: List<String>
+    ): String {
```

Comment:
(same comment)

Reply:
what do you mean by same comment here?

Reply:
confused as well

Reply:
same comment as the one above (re grantSelectSql only ever being called on users/roles)

Reply:
at this point, the parameter is already the postgres username/rolename, only a string

Reply:
lol somehow I read that as a check on principal type -- ignore me